substrate/frame/executive/src/tests.rs

```diff
@@ -44,6 +44,40 @@ use pallet_balances::Call as BalancesCall;
 
 const TEST_KEY: &[u8] = b":test:key:";
 
+#[cfg(feature = "try-runtime")]
+mod try_runtime {
+	use frame_support::traits::{IdentifiableTryStateLogic, TryStateLogic};
+	use sp_runtime::traits::AtLeast32BitUnsigned;
+
+	// Will contain `true` when the custom runtime logic is called.
+	pub(super) static mut CANARY_FLAG: bool = false;
+
+	const CUSTOM_TRY_STATE_ID: &[u8] = b"custom_try_state";
+
+	pub(super) struct CustomTryState;
+
+	impl<BlockNumber> TryStateLogic<BlockNumber> for CustomTryState
+	where
+		BlockNumber: Clone + sp_std::fmt::Debug + AtLeast32BitUnsigned,
+	{
+		fn try_state(_: BlockNumber) -> Result<(), sp_runtime::TryRuntimeError> {
+			unsafe {
+				CANARY_FLAG = true;
```

Comment:
`CANARY_FLAG::set(true);`

in combination with `parameter_types!` pub static please. Cell would also work, but normally we use parameter types.

Reply:
Addressed in https://github.com/paritytech/polkadot-sdk/pull/4631/commits/a87c5947c8c025981475c52121e5ce48c42b0eb2.